CMakeLists.txt

```diff
@@ -6,6 +6,18 @@ set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
+# Read version number from command line argument
+if(DEFINED VERSION)
+    string(REGEX MATCH "v?([0-9]+)\\.([0-9]+)\\.([0-9]+)" _ ${VERSION})
+    set(CPACK_PACKAGE_VERSION_MAJOR ${CMAKE_MATCH_1})
+    set(CPACK_PACKAGE_VERSION_MINOR ${CMAKE_MATCH_2})
+    set(CPACK_PACKAGE_VERSION_PATCH ${CMAKE_MATCH_3})
+else()
+    set(CPACK_PACKAGE_VERSION_MAJOR "0")
```

Comment:
Why set this at all if not specified? 0.0.0 seems easier to understand as the default for unset.